src/main/java/com/example/springbootbookstore/service/BookServiceImpl.java

```diff
@@ -1,28 +1,40 @@
 package com.example.springbootbookstore.service;
 
+import com.example.springbootbookstore.dto.BookDto;
+import com.example.springbootbookstore.dto.CreateBookRequestDto;
+import com.example.springbootbookstore.exception.EntityNotFoundException;
+import com.example.springbootbookstore.mapper.BookMapper;
 import com.example.springbootbookstore.model.Book;
 import com.example.springbootbookstore.repository.BookRepository;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
```

Comment:
Mapstruct supports mapping collections from the box

Reply:
I added List to mapper